File: app/src/main/java/org/skyfaced/mvp/di/NetworkModule.java
package org.skyfaced.mvp.di;

import org.skyfaced.mvp.service.network.AppNetwork;
import org.skyfaced.mvp.service.network.JikanService;
import org.skyfaced.mvp.service.network.WaifuService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public final class NetworkModule {
    @Provides
    @Singleton
    WaifuService provideWaifuService() {
        return AppNetwork.getInstance().waifuService;
    }

    @Provides
    @Singleton
    JikanService provideJikanService() {
        return AppNetwork.getInstance().jikanService;
    }
}

File: app/src/main/java/org/skyfaced/mvp/di/NetworkModule.kt
package org.skyfaced.mvp.di

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import org.skyfaced.mvp.service.network.AppNetwork
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object NetworkModule {
    @Provides
    @Singleton
    fun provideWaifuService() = AppNetwork.waifuService

    @Provides
    @Singleton
    fun provideJikanService() = AppNetwork.jikanService
}
